webpack/assets/javascripts/react_app/redux/reducers/hosts/index.js

```diff
@@ -0,0 +1,34 @@
+import {
+  HOST_POWER_STATUS_REQUEST,
+  HOST_POWER_STATUS_SUCCESS,
+  HOST_POWER_STATUS_FAILURE
+} from '../../consts';
+import Immutable from 'seamless-immutable';
+
+const initialState = Immutable({
+  powerStatus: Immutable({})
+});
+
+export default (state = initialState, action) => {
+  const { payload } = action;
+
+  switch (action.type) {
+    case HOST_POWER_STATUS_REQUEST:
+      return state.setIn(
+        ['powerStatus', payload.id],
+        payload
+      );
+    case HOST_POWER_STATUS_SUCCESS:
```

Comment:
I understand Ohad's comment but I question whether or not we what to allow breakthrough case statements. Perhaps in this case wetter may be better?